docs/design/yarn-v1.md

```diff
@@ -0,0 +1,181 @@
+
+### Installing Yarn (via Corepack)
+Yarn commands will be executed in order to populate the offline mirror, so Yarn will need to be installed in the Cachi2 image. Corepack is
+already being used to install Yarn in the Cachi2 Yarn v2+ implementation, so a similar approach is suggested.
+
+Corepack installs the latest known-good Yarn v1.x release by default if no other version is specified explicitly. It seems reasonable to always
+use the latest 1.x release for the prefetch, since 1.x is no longer under active development and we never previously allowed users to specify a specific
+Yarn version to process their request in Cachito. We could add more flexibility with a later enhancement if needed.
+
+Corepack could be configured to ignore user project configuration and use the global Yarn 1.x default by specifying the following
+[environment variable](https://github.com/nodejs/corepack#environment-variables):
+
+    COREPACK_ENABLE_PROJECT_SPEC=0
+
+Alternatively we could configure corepack explicitly using the packageManager field in package.json like is done in Cachi2's Yarn v2+ implementation.
+
+We will also need to set the following to prevent corepack from prompting for interactive user input when downloading Yarn:
+
+    COREPACK_ENABLE_DOWNLOAD_PROMPT=0
+
+### Configuring Yarn (Using Environment Variables)
+The behavior of Yarn can be configured through the use of environment variables, configuration files, and CLI options for specific commands.
+
+Environment variables will be preferred by Cachi2 when available.
+
+Unfortunately the official Yarn documentation is quite limited or non-existent concerning environment variables.
+
+The environment variables specific to the Yarn package manager are prefixed by `YARN_`. When specified, they are [loaded into the registry object](https://github.com/yarnpkg/yarn/blob/7cafa512a777048ce0b666080a24e80aae3d66a9/src/registries/base-registry.js#L125) and then made available to the rest of the Yarn application [via the Config object](https://github.com/yarnpkg/yarn/blob/7cafa512a777048ce0b666080a24e80aae3d66a9/src/config.js#L231). An [example](https://github.com/yarnpkg/yarn/blob/7cafa512a777048ce0b666080a24e80aae3d66a9/src/config.js#L422) of this can be seen for the `YARN_YARN_OFFLINE_MIRROR_PRUNING` (undocumented) environment variable.
+
+Yarn also has support for including [NPM environment variables](https://docs.npmjs.com/cli/v10/using-npm/config#environment-variables) prefixed with `NPM_CONFIG_`.
+
+### Project Configuration
+The [package.json](https://classic.yarnpkg.com/en/docs/package-json) and [yarn.lock](https://classic.yarnpkg.com/en/docs/yarn-lock)
+configuration files must be present in order to process the request.
+
+The node_modules directory should be absent. It may not be possible to guard against a user-specified modulesFolder at build-time.
+
+Yarn projects using the [Plug'n'Play (PnP)](https://classic.yarnpkg.com/lang/en/docs/pnp/) feature will not be supported.
```

Comment:
Nitpick: I'd add a very brief explanation stating we won't support it because there's no way to validate unplugged deps.

Reply:
Added :+1: